helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -245,18 +245,11 @@ public void setClusterInstanceStatus(Set<String> liveInstanceSet, Set<String> in
       // Register beans for instances that are newly configured
       Set<String> toRegister = Sets.newHashSet(instanceSet);
       toRegister.removeAll(_instanceMonitorMap.keySet());
-      Set<InstanceMonitor> monitorsToRegister = Sets.newHashSet();
       for (String instanceName : toRegister) {
-        InstanceMonitor bean = new InstanceMonitor(_clusterName, instanceName);
+        InstanceMonitor bean = getOrCreateInstanceMonitor(instanceName);
```

Comment:
This simplification might cause problem. The original design was to prevent stale instance object in the registed MBean list.
Note we treat the new instance and to be updated instance separately.

First part of your change, I think it will not change any behavior. Since we already "toRegister.removeAll(_instanceMonitorMap.keySet());".
Second part of your change is not correctly, it will cause the update fails. Basically only removing but no creating.